File: java/src/services/InventoryManager.java
package services;

import entity.Product;

public class InventoryManager {

    public void verifyStock(Product product, int quantity) {
        if (product.getStock() < quantity) {
            throw new RuntimeException("Stock insuffisant pour " + product.getName());
        }
    }

    public void deductStock(Product product, int quantity) {
        product.decreaseStock(quantity);
    }
}

File: php/services/InventoryManager.php
<?php

namespace services;

use entity\Product;

class InventoryManager
{
    public function verifyStock(Product $product, int $quantity): void
    {
        if ($product->getStock() < $quantity) {
            throw new \RuntimeException(
                "Stock insuffisant pour {$product->getName()}"
            );
        }
    }

    public function deductStock(Product $product, int $quantity): void
    {
        $product->decreaseStock($quantity);
    }
}